Zeek CI fails with latest main

The spicy tests `spicy.multiple-enum` and `spicy.tuple-enum` are currently failing in Zeek's nightly `spicy_head` run. This bumps `auxil/spicy`. It's marked may fail, but the following look rather rough:

https://cirrus-ci.com/task/6020612349493248?logs=test#L1881
```
[#3] spicy.multiple-enum ... failed
  % 'spicyz -d -o test.hlto dtest.spicy ./dtest.evt' failed unexpectedly (exit code 1)
  % cat .stderr
  ../spicy_hooks_dtest_6543fc66fc7b4bd-950f8f6a3770f5e1.cc:17:30: error: 'RESULT' is not a member of '__hlt::dtest'
     17 |         std::optional<dtest::RESULT> result_{};
        |                              ^~~~~~
  ../spicy_hooks_dtest_6543fc66fc7b4bd-950f8f6a3770f5e1.cc:17:36: error: template argument 1 is invalid
     17 |         std::optional<dtest::RESULT> result_{};
```

https://cirrus-ci.com/task/6020612349493248?logs=test#L1881
```
[#3] spicy.tuple-enum ... failed
  % 'spicyz -o test.hlto tupleenum.spicy ./tupleenum.evt' failed unexpectedly (exit code 1)
  % cat .stderr
  ../spicy_hooks_TupleEnum_3eee6cd216bdc213-4a7405a1694c56e6.cc:16:34: error: 'TestEnum' is not a member of '__hlt::TupleEnum'
     16 |         std::optional<TupleEnum::TestEnum> a{};
        |                                  ^~~~~~~~
  ../spicy_hooks_TupleEnum_3eee6cd216bdc213-4a7405a1694c56e6.cc:16:42: error: template argument 1 is invalid
     16 |         std::optional<TupleEnum::TestEnum> a{};
        |                                          ^
  ../spicy_hooks_TupleEnum_3eee6cd216bdc213-4a7405a1694c56e6.cc:31:56: error: 'TestEnum' is not a member of '__hlt::TupleEnum'
     31 |         Message(std::optional<std::optional<TupleEnum::TestEnum>> a, std::optional<std::optional<::hilti::rt::integer::safe<uint8_t>>> b);
        |                                                        ^~~~~~~~
```

Also, the CI run that installs a few Spicy analyzers fails when installing `spicy-dhcp`:

https://cirrus-ci.com/task/4894712442650624?logs=spicy_install_analyzers#L37
```
2024-08-10 03:08:34 DEBUG    building "https://github.com/zeek/spicy-dhcp": running build_command: mkdir -p build && cd build && SPICYZ=$(command -v spicyz || echo /zeek/install/var/lib/zkg/testing/spicy-dhcp/clones/spicy-plugin/build/bin/spicyz) cmake .. && cmake --build .
2024-08-10 03:08:34 INFO     installing "https://github.com/zeek/spicy-dhcp": writing build log: /zeek/install/var/lib/zkg/logs/spicy-dhcp-build.log
error: failed to run tests for https://github.com/zeek/spicy-dhcp: package build_command failed, see log in /zeek/install/var/lib/zkg/logs/spicy-dhcp-build.log
```

Unfortunately zkg swallows the actual issue.

--- a/hilti/toolchain/CMakeLists.txt
+++ b/hilti/toolchain/CMakeLists.txt
@@ -191,11 +191,11 @@ set_config_val(HILTI_CONFIG_TOOLCHAIN_CXX_LIBRARY_DIRS
 # LD flags
 set_config_val(
     HILTI_CONFIG_RUNTIME_LD_FLAGS_DEBUG
-    "${EXTRA_CXX_FLAGS} ${CMAKE_CXX_FLAGS} ${addl_ld_flags} ${EXTRA_LD_FLAGS} ${CMAKE_EXE_LINKER_FLAGS_INIT} -Wno-unused-command-line-argument"
+    "${EXTRA_CXX_FLAGS} ${CMAKE_CXX_FLAGS} ${addl_ld_flags} ${EXTRA_LD_FLAGS} ${CMAKE_EXE_LINKER_FLAGS_INIT}"
 )
 set_config_val(
     HILTI_CONFIG_RUNTIME_LD_FLAGS_RELEASE
-    "${EXTRA_CXX_FLAGS} ${CMAKE_CXX_FLAGS} ${addl_ld_flags} ${EXTRA_LD_FLAGS} ${CMAKE_EXE_LINKER_FLAGS_INIT} -Wno-unused-command-line-argument"
+    "${EXTRA_CXX_FLAGS} ${CMAKE_CXX_FLAGS} ${addl_ld_flags} ${EXTRA_LD_FLAGS} ${CMAKE_EXE_LINKER_FLAGS_INIT}"
 )
 
 configure_file(include/config.h.in ${AUTOGEN_H}/config.h)

--- a/hilti/toolchain/include/ast/type.h
+++ b/hilti/toolchain/include/ast/type.h
@@ -365,6 +365,12 @@ public:
     /** Returns true if the qualified type is constant. */
     bool isConstant() const { return _constness == Constness::Const; }
 
+    /**
+     * Returns true if the type was created through `createExternal()`. If so,
+     * `type()` will retrieve the external type.
+     */
+    auto isExternal() const { return static_cast<bool>(_external); }
+
     /** Returns the type's constness. */
     auto constness() const { return _constness; }
 

--- a/hilti/toolchain/src/ast/ast-context.cc
+++ b/hilti/toolchain/src/ast/ast-context.cc
@@ -122,6 +122,11 @@ private:
 
     void operator()(declaration::Type* n) final { insert(n); }
 
+    void operator()(QualifiedType* n) final {
+        if ( n->isExternal() )
+            follow(n->type());
+    }
+
     void operator()(expression::Name* n) final {
         auto d = n->resolvedDeclaration();
         assert(d);

--- a/hilti/toolchain/src/config.cc.in
+++ b/hilti/toolchain/src/config.cc.in
@@ -140,13 +140,14 @@ void Configuration::init(bool use_build_directory) {
     //
     // (3) it helps the optimizer to know that symbols won't be accessed
     // externally.
-    runtime_cxx_flags_debug = flatten({"-fPIC", "-std=c++17", "-g", "-fvisibility=hidden",
+    runtime_cxx_flags_debug = flatten({"-fPIC", "-std=c++17", "-g", "-fvisibility=hidden", "-Wno-invalid-offsetof",
                                        prefix("${HILTI_CONFIG_RUNTIME_CXX_INCLUDE_DIRS}", "-I", installation_tag),
                                        prefix("${HILTI_CONFIG_RUNTIME_CXX_FLAGS_DEBUG}", "", installation_tag)});
 
-    runtime_cxx_flags_release = flatten({"-fPIC", "-std=c++17", "-g", "-O3", "-DNDEBUG", "-fvisibility=hidden",
-                                         prefix("${HILTI_CONFIG_RUNTIME_CXX_INCLUDE_DIRS}", "-I", installation_tag),
-                                         prefix("${HILTI_CONFIG_RUNTIME_CXX_FLAGS_RELEASE}", "", installation_tag)});
+    runtime_cxx_flags_release =
+        flatten({"-fPIC", "-std=c++17", "-g", "-O3", "-DNDEBUG", "-fvisibility=hidden", "-Wno-invalid-offsetof",
+                 prefix("${HILTI_CONFIG_RUNTIME_CXX_INCLUDE_DIRS}", "-I", installation_tag),
+                 prefix("${HILTI_CONFIG_RUNTIME_CXX_FLAGS_RELEASE}", "", installation_tag)});
 
 
     if ( auto libhilti_pch = precompiled_libhilti(*this, true) )
@@ -155,29 +156,29 @@ void Configuration::init(bool use_build_directory) {
     if ( auto libhilti_pch = precompiled_libhilti(*this, false) )
         runtime_cxx_flags_release.push_back(rt::fmt("-include%s", libhilti_pch->c_str()));
 
-    runtime_ld_flags_debug = flatten(
-        {prefix("${HILTI_CONFIG_RUNTIME_CXX_LIBRARY_DIRS}", "-L", installation_tag),
-         prefix("${HILTI_CONFIG_RUNTIME_CXX_LIBRARY_DIRS}", "-Wl,-rpath,", installation_tag),
-         prefix("${HILTI_CONFIG_RUNTIME_LIBRARIES_DEBUG}", "-l", installation_tag),
-         prefix("${HILTI_CONFIG_RUNTIME_LD_FLAGS_DEBUG}", "", installation_tag),
-         prefix(CMAKE_THREAD_LIBS_INIT, "-l", installation_tag), prefix(CMAKE_DL_LIBS, "-l", installation_tag)});
+    runtime_ld_flags_debug =
+        flatten({prefix("${HILTI_CONFIG_RUNTIME_CXX_LIBRARY_DIRS}", "-L", installation_tag),
+                 prefix("${HILTI_CONFIG_RUNTIME_CXX_LIBRARY_DIRS}", "-Wl,-rpath,", installation_tag),
+                 prefix("${HILTI_CONFIG_RUNTIME_LIBRARIES_DEBUG}", "-l", installation_tag),
+                 prefix("${HILTI_CONFIG_RUNTIME_LD_FLAGS_DEBUG}", "", installation_tag),
+                 prefix(CMAKE_THREAD_LIBS_INIT, "-l", installation_tag), prefix(CMAKE_DL_LIBS, "-l", installation_tag),
+                 "-Wno-unused-command-line-argument"});
 
-    runtime_ld_flags_release = flatten(
-        {prefix("${HILTI_CONFIG_RUNTIME_CXX_LIBRARY_DIRS}", "-L", installation_tag),
-         prefix("${HILTI_CONFIG_RUNTIME_CXX_LIBRARY_DIRS}", "-Wl,-rpath,", installation_tag),
-         prefix("${HILTI_CONFIG_RUNTIME_LIBRARIES_RELEASE}", "-l", installation_tag),
-         prefix("${HILTI_CONFIG_RUNTIME_LD_FLAGS_RELEASE}", "", installation_tag),
-         prefix(CMAKE_THREAD_LIBS_INIT, "-l", installation_tag), prefix(CMAKE_DL_LIBS, "-l", installation_tag)});
+    runtime_ld_flags_release =
+        flatten({prefix("${HILTI_CONFIG_RUNTIME_CXX_LIBRARY_DIRS}", "-L", installation_tag),
+                 prefix("${HILTI_CONFIG_RUNTIME_CXX_LIBRARY_DIRS}", "-Wl,-rpath,", installation_tag),
+                 prefix("${HILTI_CONFIG_RUNTIME_LIBRARIES_RELEASE}", "-l", installation_tag),
+                 prefix("${HILTI_CONFIG_RUNTIME_LD_FLAGS_RELEASE}", "", installation_tag),
+                 prefix(CMAKE_THREAD_LIBS_INIT, "-l", installation_tag), prefix(CMAKE_DL_LIBS, "-l", installation_tag),
+                 "-Wno-unused-command-line-argument"});
 
     hlto_cxx_flags_debug = runtime_cxx_flags_debug;
     hlto_cxx_flags_release = runtime_cxx_flags_release;
 
     hlto_ld_flags_debug = flatten({"-shared", "-Wl,-undefined", "-Wl,dynamic_lookup",
-                                   prefix("${HILTI_CONFIG_RUNTIME_LD_FLAGS_DEBUG}", "", installation_tag),
-                                   "-Wno-unused-command-line-argument"});
+                                   prefix("${HILTI_CONFIG_RUNTIME_LD_FLAGS_DEBUG}", "", installation_tag)});
     hlto_ld_flags_release = flatten({"-shared", "-Wl,-undefined", "-Wl,dynamic_lookup",
-                                     prefix("${HILTI_CONFIG_RUNTIME_LD_FLAGS_RELEASE}", "", installation_tag),
-                                     "-Wno-unused-command-line-argument"});
+                                     prefix("${HILTI_CONFIG_RUNTIME_LD_FLAGS_RELEASE}", "", installation_tag)});
 
 #ifdef __APPLE__
     // Recent macOS versions have started to report `ld: warning: -undefined

